pkg/files/file_helpers_test.go

```diff
@@ -250,3 +254,77 @@ func TestCompareFileHash_Delete(t *testing.T) {
 		})
 	}
 }
+
+func TestDetermineFileAction(t *testing.T) {
+	addAction := mpi.File_FILE_ACTION_ADD
+	updateAction := mpi.File_FILE_ACTION_UPDATE
+	deleteAction := mpi.File_FILE_ACTION_DELETE
+	unchangedAction := mpi.File_FILE_ACTION_UNCHANGED
+
+	tempDir := os.TempDir()
+	deleteFile := tempDir + "deleteFile.conf"
+	addFile := tempDir + "addFile.conf"
+	updateFile := tempDir + "updateFile.conf"
+	unchangedFile := tempDir + "unchangedFile.conf"
+
+	modifiedFileContent := []byte("test update data")
+	currentFileContent, readErr := os.ReadFile("../../test/config/nginx/nginx.conf")
+	require.NoError(t, readErr)
+
+	currentFiles := []*mpi.File{
+		{
+			FileMeta: protos.FileMeta(deleteFile, GenerateHash(currentFileContent)),
+		},
+		{
+			FileMeta: protos.FileMeta(updateFile, GenerateHash(currentFileContent)),
+		},
+		{
+			FileMeta: protos.FileMeta(unchangedFile, GenerateHash(currentFileContent)),
+		},
+	}
+
+	modifiedFiles := []*mpi.File{
+		{
+			FileMeta: protos.FileMeta(updateFile, GenerateHash(modifiedFileContent)),
+		},
+		{
+			FileMeta: protos.FileMeta(unchangedFile, GenerateHash(currentFileContent)),
+		},
+		{
+			FileMeta: protos.FileMeta(addFile, GenerateHash(modifiedFileContent)),
+		},
+	}
+
+	expectedResult := []*mpi.File{
+		{
+			FileMeta: protos.FileMeta(deleteFile, GenerateHash(currentFileContent)),
+			Action:   &deleteAction,
+		},
+		{
+			FileMeta: protos.FileMeta(updateFile, GenerateHash(modifiedFileContent)),
+			Action:   &updateAction,
+		},
+		{
+			FileMeta: protos.FileMeta(unchangedFile, GenerateHash(currentFileContent)),
+			Action:   &unchangedAction,
+		},
+		{
+			FileMeta: protos.FileMeta(addFile, GenerateHash(modifiedFileContent)),
+			Action:   &addAction,
+		},
+	}
+
+	result := DetermineFileAction(currentFiles, modifiedFiles)
+
+	slices.SortFunc(result, func(a, b *mpi.File) int {
+		return cmp.Compare(a.GetFileMeta().GetName(), b.GetFileMeta().GetName())
+	})
+
+	slices.SortFunc(expectedResult, func(a, b *mpi.File) int {
+		return cmp.Compare(a.GetFileMeta().GetName(), b.GetFileMeta().GetName())
+	})
+
```

Comment:
Can you add an assert to make sure the number of files in the `result` is equal to the number files in the `expectedResult`